### PR TITLE
Attr agent conf

### PR DESCRIPTION
**Pull request by Zibby**

### CODE REVIEW

**Comment by jamesyale** [comment]
You've not explained what this is meant to achieve.

**Comment by essjayhch** [comment]
its so you can override the server and server active settings via node attributes

File: resources/agent.rb [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
 property :enable_remote_commands, [0, 1], default: 0
 property :log_remote_commands, [0, 1], default: 1
-property :server, String, default: 'DYNAMIC'
+property :server, String, default: node['zabbix']['agent']['server']
 property :listen_port, Integer, default: 10_050
 property :listen_ip, String, default: node['ipaddress']
 property :start_agents, Integer, default: 3
-property :server_active, String, default: 'DYNAMIC'
-property :hostname, String, default: node['fqdn']
+property :active, String, default: node['zabbix']['agent']['active']
```
Comment (essjayhch) [inline resources/agent.rb]: Active what? Agent? 

This sets the ServerActive configuration directive, as such server_active is entirely appropriate within the ruby coding conventions.

further;

Don't arbitrarily change what is effectively an external resource consumed by other code unless there is a very good reason.

Comment (Zibby) [reply resources/agent.rb]: agreed. sorry was trying to get around that linting issue, ill fix that now

**Comment by Zibby** [comment]
> You've not explained what this is meant to achieve.

This is mostly to get zfs box working with new zbx, but can be used on other hosts. 
Allows for zbx host/ proxy to be manually assigned and the hostname to be overridden using node attributes. Eg zfs-01.mch.livelink.io keeps auto assigning itself to zfs-01.dmz.livelink.io and going into alert.

**Pull request merged**